tamr_toolbox/data_io/cloud.py

```diff
@@ -0,0 +1,220 @@
+import tempfile
+import tarfile
+
+
+def gcs_upload(
+    cloud_client,
```

Comment:
This and the rest of the arguments should be correctly type-hinted.
You can `from google.cloud.storage import Client as GcsClient` at the top of the file, then
```suggestion
    cloud_client: "GcsClient",
```
The type needs to be quoted since this is an optional dependency.